File: CS/App_Code/CustomFileProvider.cs
using DevExpress.Web;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Web;

public class CustomFileProvider : PhysicalFileSystemProvider {
	public CustomFileProvider(string rootFolder)
		: base(rootFolder) { }
	
	public override IEnumerable<FileManagerFolder> GetFolders(FileManagerFolder parentFolder) {
		List<FileManagerFolder> folders = new List<FileManagerFolder>(base.GetFolders(parentFolder));
		return folders.OrderByDescending(a => a.Name);
	}
}

File: VB/App_Code/CustomFileProvider.vb
Imports DevExpress.Web
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Web

Public Class CustomFileProvider
    Inherits PhysicalFileSystemProvider

    Public Sub New(ByVal rootFolder As String)
        MyBase.New(rootFolder)
    End Sub
    
    Public Overrides Function GetFolders(ByVal parentFolder As FileManagerFolder) As IEnumerable(Of FileManagerFolder)
        Dim folders As New List(Of FileManagerFolder)(MyBase.GetFolders(parentFolder))
        Return folders.OrderByDescending(Function(a) a.Name)
    End Function
End Class
